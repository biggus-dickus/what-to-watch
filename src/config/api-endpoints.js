export const BASE_HOST = `https://es31-server.appspot.com`;

export const ApiEndpoint = {
  BASE_URL: `${BASE_HOST}/wtw`,
  COMMENTS: `/comments`,
  FILMS: `/films`,
  FAVOURITE: `favorite`,
  LOGIN: `/login`
};
