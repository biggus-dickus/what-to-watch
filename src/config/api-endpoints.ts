export const BASE_HOST: string = `https://es31-server.appspot.com`;

export enum ApiEndpoint {
  BASE_URL = 'https://es31-server.appspot.com/wtw',
  COMMENTS = '/comments',
  FILMS = '/films',
  FAVOURITE = 'favorite',
  LOGIN = '/login',
  PROMO = '/films/promo'
}
