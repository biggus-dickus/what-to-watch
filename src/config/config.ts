// How much time to wait in ms before playing a film preview
export const HOVER_DELAY = 1000;

// How many films to render in "More like this" section (film page)
export const MORE_LIKE_THIS_LIMIT = 4;

// How many films to display at once on mount
export const FILMS_LIMIT = 20;

// How many genres to display at once
export const GENRES_LIMIT = 9 + 1; // 9 + "All genres"

// Navigation tabs on film page
export const NAV_OVERVIEW_ID = `overview`;
export const NAV_DETAILS_ID = `details`;
export const NAV_REVIEWS_ID = `reviews`;
export const filmTabs = [
  {
    id: NAV_OVERVIEW_ID,
    text: `Overview`
  },
  {
    id: NAV_DETAILS_ID,
    text: `Details`
  },
  {
    id: NAV_REVIEWS_ID,
    text: `Reviews`
  }
];

// Film review constraints
export const REVIEW_CHARS_MIN = 50;
export const REVIEW_CHARS_MAX = 400;
