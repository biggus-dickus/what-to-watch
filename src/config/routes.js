const Route = Object.freeze({
  ADD_REVIEW: `/film/:id/review`,
  FILM: `/film/:id`,
  INDEX: `/`,
  MY_LIST: `/mylist`,
  SIGN_IN: `/login`
});

export default Route;

export const isPrivateRoute = (pathname) => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[\d]\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
