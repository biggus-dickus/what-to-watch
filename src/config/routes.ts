enum Route {
  ADD_REVIEW = '/film/:id/review',
  FILM = '/film/:id',
  INDEX = '/',
  MY_LIST = '/mylist',
  SIGN_IN = '/login'
}

export default Route;

export const isPrivateRoute = (pathname: string): boolean => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[1-9][\d]*\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
