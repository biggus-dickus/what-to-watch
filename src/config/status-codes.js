export const StatusCode = Object.freeze({
  OK: 200,
  NOT_MODIFIED: 304,
  BAD_REQUEST: 400,
  FORBIDDEN: 403,
  NOT_FOUND: 404,
  INTERNAL_SERVER_ERROR: 500,
  SERVICE_UNAVAILABLE: 503
});
