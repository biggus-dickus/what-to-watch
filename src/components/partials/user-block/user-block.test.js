import React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {BASE_HOST} from '../../../config/api-endpoints';
import UserBlock from './user-block';

configure({adapter: new Adapter()});

const mockUser = {
  avatar_url: '/wtw/static/avatar/10.jpg', // eslint-disable-line
  id: 777,
  email: `test@test.com`,
  name: `Poehavshiy`,
};


describe(`UserBlock test suite.`, () => {
  it(`<UserBlock /> should render correctly`, () => {
    const tree = renderer.create(<UserBlock user={mockUser} onLinkClick={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should contain a link to sign-in page if a user is not logged in`, () => {
    const wrapper = shallow(<UserBlock user={null} onLinkClick={jest.fn()} />);

    expect(wrapper.find(`a`)).toHaveLength(1);
  });

  it(`should show the user's avatar if they are logged in`, () => {
    const wrapper = shallow(<UserBlock user={mockUser} onLinkClick={jest.fn()} />);

    expect(wrapper.find(`img`).prop(`src`)).toEqual(`${BASE_HOST}${mockUser.avatar_url}`);
  });
});
