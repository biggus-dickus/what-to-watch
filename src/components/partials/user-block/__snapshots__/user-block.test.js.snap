// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock test suite. <UserBlock /> should render correctly 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__avatar"
    href="/mylist"
    onClick={[Function]}
  >
    <img
      alt="Nastasha Romanenko"
      height="63"
      src="https://es31-server.appspot.com/wtw/static/avatar/10.jpg"
      width="63"
    />
  </a>
</div>
`;
